pkg/ebpf/c/bpf_metadata.h

```diff
@@ -0,0 +1,19 @@
+/**
+ * This header file is included in kconfig.h and ktypes.h, which should make
+ * this metadata available to all object files.
+ */
+
+#ifndef __BPF_METADATA_H__
+#define __BPF_METADATA_H__
+
+#ifndef __CGO__ // Do not include metadata in CGo builds, as it causes duplicate symbols
+#if defined(__x86_64__) || defined(__TARGET_ARCH_x86)
+char __dd_metadata_arch[] __attribute__((section("dd_metadata"), used)) = "<arch:x86_64>";
+#elif defined(__aarch64__) || defined(__TARGET_ARCH_arm64)
+char __dd_metadata_arch[] __attribute__((section("dd_metadata"), used)) = "<arch:arm64>";
+#else
+char __dd_metadata_arch[] __attribute__((section("dd_metadata"), used)) = "<arch:unset>";
```

Comment:
What do you think about removing this, since we only support `arm64` and `x86_64`.
Moreover, we can do a build time check in `system-probe.py` for the existence of these sections and failing early, when they are not present. That way devs don't have to wait till the e2e check fails in case of incorrect build.

Reply:
The build time check sounds like a great idea, I will add them in a followup PR.

About removing the `unset` metadata, I wanted to have something to distinguish files with metadata but a wrong arch from files without metadata. Basically a sanity check for me because sometimes I wasn't sure if I was adding the metadata incorrectly or just looking at old files that weren't updated properly.